x-pack/plugins/enterprise_search/server/lib/route_config_helpers.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { RouteConfig, RouteConfigOptionsBody, RouteMethod } from 'kibana/server';
+
+import { schema } from '@kbn/config-schema';
+
+/**
+ * Kibana Enterprise Search Plugin API endpoints often times pass through the request
+ * body to the Enterprise Search API endpoints for validation. In those cases, we do not
+ * need to validate them in Kibana.
+ *
+ * The safe way to do that is to turn off body parsing entirely using `options.body.parse: false`.
+ * The will pass a String Buffer to the route handler. The proper way to validate this when validation
+ * is enabled to to use `body: schema.buffer()`.
+ *
+ * Example:
+ *  router.put({
+ *    path: '/api/app_search/engines/{engineName}/example',
+ *    validate: {
+ *      params: schema.object({
+ *        engineName: schema.string(),
+ *      }),
+ *      body: schema.buffer(),
+ *    },
+ *    options: { body: { parse: false } },
+ *  },
+ *  ...
+ *
+ * This helper applies that pattern, while maintaining existing options:
+ *
+ *  router.put(skipBodyValidation({
+ *    path: '/api/app_search/engines/{engineName}/example',
+ *    validate: {
+ *      params: schema.object({
+ *        engineName: schema.string(),
+ *      }),
+ *    },
+ *  },
+ *  ...
+ */
+export const skipBodyValidation = <Method extends RouteMethod>(
+  config: RouteConfig<unknown, unknown, unknown, Method>
+): RouteConfig<unknown, unknown, unknown, Method> => {
+  const options = config.options || {};
+  const bodyOptions = config.options?.body || {};
+
+  // We throw here rather than overwriting the existing settings to avoid confusion
+  if (config.validate && config.validate.body) {
+    throw new Error('validate.body cannot be set when using "skipBodyValidation"');
+  }
+  if (options.body) {
+    throw new Error('options.body cannot be set when using "skipBodyValidation"');
+  }
+
+  return {
+    ...config,
+    validate:
+      config.validate === false
+        ? false
+        : {
+            ...config.validate,
+            body: schema.buffer(),
+          },
+    options: {
+      ...options,
+      body: {
+        ...bodyOptions,
+        parse: false,
+        // This is casted explicitly to the expected type to work around TS errors.
+        // skipBodyValidation should never be used on 'get' or 'options' requests which
+        // would have no body
+      } as Method extends 'get' | 'options' ? undefined : RouteConfigOptionsBody,
```

Comment:
Is the method available anywhere in the `config` blob? If so, maybe we should just throw on it as well as opposed to enforcing via Typescript? 🤔 

No worries if it's not though, TS enforcing is also just fine - more of a consistency thing that occurred to me. Now I'm wondering if we could enforce our earlier throws via TS as well - maybe `config.validate.body: never` and `config.options.body: never`? Just rambling out loud now

Reply:
Yeah, probably. If it were our own types I wouldn't hesitate, the Kibana types for Router just make my head spin. I'll spend a little time trying to figure it out this morning.

Reply:
Haha sorry I'd say don't worry about it then, super did not mean to send you down a rabbit hole with this!

Reply:
Ugh, I got about this far on the validations and then gave up. I have a partial fix I'll push up.

```
interface ConfigWithoutBodyOptions<Method extends DestructiveRouteMethod> extends Config<Method> {
  validate: Omit<Config<Method>['validate'], 'body'>;
  options?: Omit<Config<Method>['options'], 'body'>;
}
```

Reply:
TS now disallows use on `get` and `options` methods. I also simplified the type signatures a bit: e3dda78fd6d91092c815f3c566191f506e4a734d

Reply:
OK, nm, I got this working... 

```ts
interface ConfigWithoutBodyOptions<P, Q, B, Method extends RouteMethod>
  extends RouteConfig<P, Q, B, Method> {
  validate: Omit<RouteValidatorFullConfig<P, Q, B>, 'body'> | false;
  options?: Omit<RouteConfigOptions<Method>, 'body'>;
}
```

Will push something up

Reply:
Check this out: 5c3609de20db5b91e75fba66f82f1e5173a4df5a

Reply:
Dang!!!! That's cool as heck, I love it!!! Thanks so much for going above and beyond Jason! 🎉